File: PCMC/Entities/Project.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace PCMC.Entities
{
    /**
     * Class entity model for projects. Entity Framework will generate a schema using code first approach.
     */
    public class Project
    { 
        public int ID { get; set; }                 //DB key for the project.
        public byte[] RawZipFileJudges { get; set; }//The zip file for judge instructions
        public byte[] RawZipFileParticipants { get; set; }//The zip file of participant instructions
        public string Name { get; set; }            // A Friendly name for the project .. Project#1 - LinkedLists
        public string Description { get; set; }     // A short description of the project ... testing knowledge of LinkedLists..
        public int MaxScore { get; set; }           // The maximum score participants may obtain (In general a score is bounded [-1,MaxScore] where -1 represents ungraded
        public bool Hidden { get; set; }            // Whether this project is hidden from participants
        public Level Level { get; set; }            // The level associated with the project
    }
}

File: PCMC/Client/Common/POCO/Project.ts
/*
 *  A Project class used to retain information sent from the server
 */
class Project {
    public constructor(ID: number, JudgeZip: string, ParZip: string, Name: string, MaxScore: number, Desc: string, IsHidden: boolean
        , Level: Level) {
        this.ID = ID;
        this.RawZipFileJudges = JudgeZip;
        this.RawZipFileParticipants = ParZip;
        this.Name = Name;
        this.MaxScore = MaxScore;
        this.Description = Desc;
        this.Hidden = IsHidden;
        this.Level = Level;
    }

    public ID: number;
    public RawZipFileJudges: string;
    public RawZipFileParticipants: string;
    public Name: string;
    public MaxScore: number;
    public Description: string;
    public Hidden: boolean;
    public Level: Level;
}
